index.js

```diff
@@ -0,0 +1,114 @@
+import assert from 'node:assert'
+import ms from 'ms'
+
+export class CancelStuckTransactions {
+  #store
+  #list
+  #resolve
+  #log
+  #sendTransaction
+  constructor ({ store, list, resolve, log, sendTransaction }) {
+    this.#store = store
+    this.#list = list
+    this.#resolve = resolve
+    this.#log = log
+    this.#sendTransaction = sendTransaction
+  }
+
+  async pending (tx) {
+    assert.strictEqual(typeof tx.hash, 'string')
+    assert.strictEqual(typeof tx.from, 'string')
+    assert.strictEqual(typeof tx.maxPriorityFeePerGas, 'bigint')
+    assert.strictEqual(typeof tx.nonce, 'number')
+    await this.#store({
+      ...tx,
+      timestamp: new Date().toISOString()
+    })
+  }
+
+  async successful (tx) {
+    assert.strictEqual(typeof tx.hash, 'string')
+    await this.#resolve(tx.hash)
+  }
+
+  async olderThan (ageMs) {
+    assert.strictEqual(typeof ageMs, 'number')
+
+    this.#log('Checking for stuck transactions...')
+    const txs = await this.#list()
+    const txsToCancel = txs.filter(tx => {
+      return new Date() - new Date(tx.timestamp) > ageMs
+    })
+    if (txsToCancel.length === 0) {
+      this.#log('No transactions to cancel')
+      return
+    }
+
+    this.#log('Transactions to cancel:')
+    for (const tx of txsToCancel) {
+      this.#log(
+        `- ${tx.hash} (age ${ms(new Date() - new Date(tx.timestamp))})`
```

Comment:
Please include the nonce in the log message. I found it very useful while cancelling the >400 stuck messages.